File: AOJ/ITP1/ITP1_10_D_Distance_II.py
n = int(input())
x = list(map(int, input().split()))
y = list(map(int, input().split()))
dxy = [abs(a - b) for a, b in zip(x, y)]
D = [sum([n ** p for n in dxy]) ** (1 / p) for p in range(1, 4)]
D.append(max(dxy))
for d in D:
    print("{:.5f}".format(d))

File: AOJ/ITP1/ITP1_10_D_Distance_II.cs
using System;
using System.Linq;
using static System.Console;

class P
{
    static void Main()
    {
        ReadLine();
        var x = ReadLine().Split().Select(int.Parse);
        var y = ReadLine().Split().Select(int.Parse);
        var d = x.Zip(y, (a, b) => Math.Abs(a - b));
        for (int i = 1; i < 4; i++)
            WriteLine(Math.Pow(d.Select(z => Math.Pow(z, i)).Sum(), 1.0 / i));
        WriteLine(d.Max());
    }
}
